packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2710,49 +2710,88 @@ class _Location {
   }
 }
 
-/// Describer to gather important information about [offendingContext].
+/// Transformer to parse and gather information about [offendingContext].
 ///
-/// This describer is only applicable when the [offendingContext] is created in
-/// widget layer.
+/// This transformer only responds to [DiagnosticsProperty] with the name
+/// 'offendingContext'.
 ///
-/// This function will be registered to [FlutterErrorDetails.offendingContextDescribers]
+/// This function will be registered to [FlutterErrorDetails.informationTransformers]
 /// in [WidgetsBinding.initInstances].
-Iterable<DiagnosticsNode> describeOffendingContext(dynamic offendingContext) {
+DiagnosticPropertiesBuilder transformOffendingContext(DiagnosticPropertiesBuilder properties) {
+  final DiagnosticPropertiesBuilder transformed = DiagnosticPropertiesBuilder();
+  bool foundStackTrace = false;
```

Comment:
this code will be a lot more readable when tweaked to use List<DiagnosticsNode> instead of DiagnosticPropertiesBuilder.
List has helper methods that make a lot of this code cleaner to write.

Reply:
I think this is currently the most efficient way to go through the list, it only go through the list at most twice.

Reply:
the refactor using yield looks a lot nicer.